extended/src/main/java/io/kubernetes/client/extended/workqueue/ratelimiter/BucketRateLimiter.java

```diff
@@ -0,0 +1,56 @@
+package io.kubernetes.client.extended.workqueue.ratelimiter;
+
+import io.github.bucket4j.*;
+import java.time.Duration;
+
+/** A light-weight token bucket implementation for RateLimiter. */
+public class BucketRateLimiter<T> implements RateLimiter<T> {
```

Comment:
thanks, the new bucketing library looks compelling. i tried simplify implemtation of `when` a bit learning from https://github.com/vladimir-bukhtoyarov/bucket4j/issues/96, wdyt?
 
```suggestion
public class BucketRateLimiter<T> implements RateLimiter<T> {

  private Bucket bucket;
  private long tokensInQueue;
  private long tokensGeneratedInPeriod;
  private Duration period;

  /**
   * @param capacity                Capacity is the maximum number of tokens can be consumed.
   * @param tokensGeneratedInPeriod Tokens generated in period.
   * @param period                  Period that generating specific number of tokens.
   */
  public BucketRateLimiter(long capacity, long tokensGeneratedInPeriod, Duration period) {
    Bandwidth bandwidth =
            Bandwidth.classic(capacity, Refill.greedy(tokensGeneratedInPeriod, period));

    this.bucket = Bucket4j.builder().
            addLimit(bandwidth).
            withSynchronizationStrategy(SynchronizationStrategy.SYNCHRONIZED).
            build();
  }

  @Override
  public synchronized Duration when(T item) {
    DelayGetter delayGetter = new DelayGetter();
    bucket.asAsyncScheduler().consume(1, delayGetter).complete(null);
    return delayGetter.getDelay();
  }

  @Override
  public void forget(T item) {
  }

  @Override
  public int numRequeues(T item) {
    return 0;
  }

  private class DelayGetter extends ScheduledThreadPoolExecutor {
    private Duration delay = Duration.ZERO;

    @Override
    public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
      this.delay = Duration.ofNanos(unit.toNanos(delay));
      return null;
    }

    public DelayGetter() {
      super(0);
    }

    public Duration getDelay() {
      return delay;
    }
  }
}

```

Reply:
oops, just realized my implementation is wrong :(

I will replace with your implementation, thanks!


withSynchronizationStrategy(SynchronizationStrategy.SYNCHRONIZED).
 